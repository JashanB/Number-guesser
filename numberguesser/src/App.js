import React, { useState, useEffect } from "react";
import logo from './logo.svg';
import './App.css';
import Form from './Form'
import NumberSetter from './NumberSetter'
import Result from './Result'

function App() {
  const [inputNumber, setInputNumber] = useState(0);
  const [range, setRange] = useState(10);
  const [compareAnswer, setCompareAnswer] = useState(false);
  const [actualAnswer, setActualAnswer] = useState(-1);
  const [checkAnswer, setCheckAnswer] = useState(false);
  const [restart, setRestart] = useState(false);

  const makeRandomNumber = function (max) {
    if (max > 1) {
      return Math.floor(Math.random() * Math.floor(max));
    } else {
      return Math.round(Math.random());
    }
  };

  useEffect(() => {
    let newNumber = makeRandomNumber(range);
    console.log(restart)
    setActualAnswer(state => newNumber);
  }, [range, restart]);

  return (
    <div className="App">
      <h2>Need a break?</h2>
      <h3> Guess a number out of {range}</h3>
      <Form
        checkAnswer={checkAnswer}
        setCheckAnswer={setCheckAnswer}
        actualAnswer={actualAnswer}
        compareAnswer={compareAnswer}
        setCompareAnswer={setCompareAnswer}
        inputNumber={inputNumber}
        setInputNumber={setInputNumber}
      />
      <NumberSetter
        range={range}
        setRange={setRange}
      />
      {checkAnswer &&
        <h4>Answer:</h4> &&
        <Result
          restart={restart}
          setRestart={setRestart}
          actualAnswer={actualAnswer}
          compareAnswer={compareAnswer}
          setCheckAnswer={setCheckAnswer}
          range={range}
          setRange={setRange}
          setInputNumber={setInputNumber}
        />}
    </div>
  );
}

export default App;
