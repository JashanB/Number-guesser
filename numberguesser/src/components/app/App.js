import React, { useState, useEffect } from "react";
import './App.css';
import Form from '../form';
import NumberSetter from '../numberSetter';
import Result from '../result';

function App() {
  const [inputNumber, setInputNumber] = useState(0);
  const [range, setRange] = useState(10);
  const [compareAnswer, setCompareAnswer] = useState(false);
  const [actualAnswer, setActualAnswer] = useState(-1);
  const [checkAnswer, setCheckAnswer] = useState(false);
  const [restart, setRestart] = useState(false);

  const makeRandomNumber = function (max) {
    if (max > 1) {
      return Math.floor(Math.random() * Math.floor(max));
    } else {
      return Math.round(Math.random());
    }
  };

  useEffect(() => {
    let newNumber = makeRandomNumber(range);
    setActualAnswer(state => newNumber);
  }, [range, restart]);

  return (
    <div className="App" data-testid="app">
      <h2>Need a break?</h2>
      <h3> Guess a number out of {range}</h3>
      <Form
        checkAnswer={checkAnswer}
        setCheckAnswer={setCheckAnswer}
        actualAnswer={actualAnswer}
        compareAnswer={compareAnswer}
        setCompareAnswer={setCompareAnswer}
        inputNumber={inputNumber}
        setInputNumber={setInputNumber}
      />
      <NumberSetter
        range={range}
        setRange={setRange}
      />
      {checkAnswer &&
        <h4>Answer:</h4> &&
        <Result
          restart={restart}
          setRestart={setRestart}
          actualAnswer={actualAnswer}
          compareAnswer={compareAnswer}
          setCheckAnswer={setCheckAnswer}
          range={range}
          setRange={setRange}
          setInputNumber={setInputNumber}
        />}
    </div>
  );
}

export default App;
