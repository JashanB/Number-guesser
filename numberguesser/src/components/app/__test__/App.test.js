import React from "react";
import ReactDOM from 'react-dom';
import App from './../App';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App></App>, div);
});

it ("renders correctly", () => {
  const {getByTestId} = render(<App></App>);
  expect(getByTestId('app')).toHaveTextContent("Need a break?");
});

it("matches snapshot", () => {
  const tree = renderer.create(<App></App>).toJSON();
  expect(tree).toMatchSnapshot();
});