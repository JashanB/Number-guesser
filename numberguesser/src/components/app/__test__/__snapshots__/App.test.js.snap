// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="app"
>
  <h2>
    Need a break?
  </h2>
  <h3>
     Guess a number out of 
    10
  </h3>
  <div
    className="input-your-guess"
  >
    <form
      autoComplete="off"
      data-testid="input-form"
      onSubmit={[Function]}
    >
      <input
        className="guess-number"
        data-testid="number-input"
        onChange={[Function]}
        placeholder="Guess a number!"
        type="number"
        value={0}
      />
    </form>
    <button
      className="check-button"
      data-testid="check-button"
      onClick={[Function]}
    >
      Check!
    </button>
  </div>
  <div
    className="set-your-number"
  >
    <h3>
      Change the range: 
    </h3>
    <form
      autoComplete="off"
      onSubmit={[Function]}
    >
      <input
        className="number-setter"
        data-testid="number-setter-form"
        onChange={[Function]}
        placeholder="Guess a number!"
        type="number"
        value={10}
      />
    </form>
  </div>
</div>
`;
